warehouse/snowflake/snowflake_test.go

```diff
@@ -47,121 +37,109 @@ func (t *TestHandle) VerifyConnection() error {
 		return err
 	}
 
-	return testhelper.WithConstantBackoff(func() (err error) {
-		caseSensitiveCredentials := credentials
-		caseSensitiveCredentials.DBName = strings.ToLower(credentials.DBName)
-		handle.CaseSensitiveDB, err = snowflake.Connect(caseSensitiveCredentials)
-		if err != nil {
-			err = fmt.Errorf("could not connect to warehouse snowflake with error: %w", err)
-			return
-		}
-		return
-	})
 }
 
-func TestSnowflakeIntegration(t *testing.T) {
-	t.SkipNow()
 
-	t.Cleanup(func() {
-		require.NoError(t, testhelper.WithConstantBackoff(func() (err error) {
-			_, err = handle.DB.Exec(fmt.Sprintf(`DROP SCHEMA "%s" CASCADE;`, handle.Schema))
-			return
-		}), fmt.Sprintf("Failed dropping schema %s for Snowflake", handle.Schema))
-	})
+func TestMain(m *testing.M) {
+	//_, exists := os.LookupEnv(testhelper.SnowflakeIntegrationTestCredentials)
+	//if !exists {
+	//	log.Println("Skipping Snowflake Test as the Test credentials does not exists.")
+	//	return
+	//}
+	//
+	//handle = &TestHandle{
+	//}
+	// 
+	//os.Exit(testhelper.Run(m, handle))
+}
 
-	t.Run("Normal Database", func(t *testing.T) {
-		t.Parallel()
 
-		warehouseTest := &testhelper.WareHouseTest{
-			Client: &client.Client{
-				SQL:  handle.DB,
-				Type: client.SQLClient,
-			},
-			WriteKey:      handle.WriteKey,
-			Schema:        handle.Schema,
-			Tables:        handle.Tables,
-			Provider:      warehouseutils.SNOWFLAKE,
-			SourceID:      "24p1HhPk09FW25Kuzvx7GshCLKR",
-			DestinationID: "24qeADObp6eIhjjDnEppO6P1SNc",
-		}
+func TestSnowflakeIntegration(t *testing.T) {
+	t.SkipNow()
+
+	credentials, err := testhelper.SnowflakeCredentials()
+	if err != nil {
+		return err
+	}
+	
+	testcase := []struct {
+		name string
+		dbName string
+		schema string
+	}{
```

Comment:
Using table test to avoid code repetition, but also to highlight the attributes we are testing every time

Reply:
Yes.